coap-core/src/main/java/com/mbed/coap/transport/udp/DatagramSocketTransport.java

```diff
@@ -160,13 +159,16 @@ public void sendPacket0(CoapPacket coapPacket, InetSocketAddress adr, TransportC
         socket.send(datagramPacket);
     }
 
-    @Override
+    protected boolean socketCreated() {
+        return (socket != null);
+    }
+
     public InetSocketAddress getLocalSocketAddress() {
-        return (InetSocketAddress) socket.getLocalSocketAddress();
+        return (InetSocketAddress) getSocket().getLocalSocketAddress();
     }
 
-    protected DatagramSocket getSocket() {
-        return socket;
+    public DatagramSocketAdapter getSocket() {
```

Comment:
Change return type to `BlockingSocket`, it will not always be `DatagramSocketAdapter`.

Reply:
fixed.